ts/screens/wallet/payment/ManualDataInsertionScreen.tsx

```diff
@@ -155,6 +157,21 @@ class ManualDataInsertionScreen extends React.Component<Props, State> {
     );
   };
 
+  /**
+   * Converts the validator state into a color string.
+   * @param isFieldValid - the validator state.
+   * @returns green string if isFieldValid is true, red string if false, undefined if undefined.
+   */
+  private getColorFromInputValidatorState(
+    isFieldValid: boolean | undefined
+  ): string | undefined {
+    return isFieldValid === undefined
+      ? undefined
+      : isFieldValid
+      ? IOColors.green
+      : IOColors.red;
+  }
```

Comment:
```suggestion
   private getColorFromInputValidatorState(
    isFieldValid: boolean | undefined
  ){
    if (isFieldValid === undefined) {
      return undefined;
    }
    return isFieldValid ? IOColors.green : IOColors.red;
  }
```
I believe this would make the function more readable, what do you think about it?
Also there is no need to define the return type since it will be inferred.

Reply:
While I agree that chained ternary operators might not be easily readable at first glance, I still prefer them over multiple return points, when possible. 

However, I took the chance and rewrote it by using fp-ts. Does it look any better now?